`showRenderTreeForThis` should dump information about content-visibility

rwlbuis opened the pull request:
#### abab8aa0b2236bbb3af49417ea05679df2389a0e
<pre>
`showRenderTreeForThis` should dump information about content-visibility
<a href="https://bugs.webkit.org/show_bug.cgi?id=261999">https://bugs.webkit.org/show_bug.cgi?id=261999</a>

Reviewed by Alan Baradlay and Tim Nguyen.

Provide debug information on hidden/auto content-visibility and skipped content.

* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::outputRenderTreeLegend):
(WebCore::RenderObject::outputRenderObject const):

Canonical link: <a href="https://commits.webkit.org/268411@main">https://commits.webkit.org/268411@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/de017e450e54e8ae7e016c3fe0016f68a07685cb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19641 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20062 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20674 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21533 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18367 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19878 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23323 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20203 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19956 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19859 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19871 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17079 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22387 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17053 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17870 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/24168 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18119 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18044 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22146 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18649 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15810 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17796 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4689 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22151 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18475 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderObject.cpp:
```diff
@@ -1337,6 +1337,14 @@ void RenderObject::outputRenderObject(TextStream& stream, bool mark, int depth)
+            if (style().contentVisibility() == ContentVisibility::Hidden)
+                stream << " [content-visibility=hidden]";
+            else
+                stream << " [content-visibility=auto]";
```
dump lines are getting really long now. I'd prefer it as part of the annotation column.

rwlbuis replied:
I like that idea, new output is:
```
(B)lock/(I)nline/I(N)line-block, (A)bsolute/Fi(X)ed/(R)elative/Stic(K)y, (F)loating, (O)verflow clip, Anon(Y)mous, (G)enerated, has(L)ayer, hasLayer(S)crollableArea, (C)omposited, Content-visibility:(H)idden/(A)uto, (S)kipped content, (+)Dirty style, (+)Dirty layout
B---YGLSC-- --* RenderView at (0,0) size 800x600 renderer (0x162000ab0) layout box (0x0)
B-----LS--- --    HTML RenderBlock at (0,0) size 800x166 renderer (0x162006920) layout box (0x0) node (0x16201a7a0)
B---------- --      BODY RenderBody at (8,8) size 784x150 renderer (0x162006b90) layout box (0x0) node (0x16201b050)
B--O--L--H- --        DIV RenderBlock at (0,0) size 150x150 renderer (0x162006e00) layout box (0x0) node (0x162005240)
B---YG----S --          RenderBlock at (0,0) size 0x0 renderer (0x162007870) layout box (0x0)
I---------S --            #text RenderText renderer (0x162005d60) layout box (0x0) node (0x1620051c0) length->(11) "\n  Text.\n  "
B---------S --          DIV RenderBlock at (0,0) size 0x0 renderer (0x162007600) layout box (0x0) node (0x1620052d0)
B---YG----S --          RenderBlock at (0,0) size 0x0 renderer (0x162005e20) layout box (0x0)
I---------S --            SPAN RenderInline renderer (0x1620060a0) layout box (0x0) node (0x162005460)
I---------S --              #text RenderText renderer (0x1620062c0) layout box (0x0) node (0x1620053e0) length->(12) "inline child"
I---------S --            #text RenderText renderer (0x162006380) layout box (0x0) node (0x162005580) length->(3) "\n  "
BA----L---S --            DIV RenderBlock at (0,0) size 0x0 renderer (0x162005f60) layout box (0x0) node (0x162005600)
BR----L---S --          DIV RenderBlock at (10,10) size 0x0 renderer (0x162006180) layout box (0x0) node (0x162005690)
BX----L---S --          DIV RenderBlock at (0,0) size 0x0 renderer (0x162006440) layout box (0x0) node (0x162005820)
BR----L---S --          DIV RenderBlock at (0,0) size 0x0 renderer (0x1620066b0) layout box (0x0) node (0x1620058b0)
BA----L---S --            DIV RenderBlock at (0,0) size 0x0 renderer (0x162001d80) layout box (0x0) node (0x162005a40)

alanbaradlay replied:
yeah this looks much better

rwlbuis replied:
> yeah this looks much better

Thanks! I'll add Tim since he suggested doing this in the first place.

nt1m replied:
Seems to contain all the needed info. Looks good to me.

rwlbuis committed:
`showRenderTreeForThis` should dump information about content-visibility
https://bugs.webkit.org/show_bug.cgi?id=261999

Reviewed by Alan Baradlay and Tim Nguyen.

Provide debug information on hidden/auto content-visibility and skipped content.

* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::outputRenderTreeLegend):
(WebCore::RenderObject::outputRenderObject const):

Canonical link: https://commits.webkit.org/268411@main
